.github/workflows/linting.yml

```diff
@@ -1,6 +1,8 @@
 name: Run Linters
 
-on: [push]
+on:
+  - push
+  - pull_request
```

Comment:
likewise, `pull_request` might need to be restricted to specify which type should run the linters. `open` is obvious, but I'm not sure how to get it to re-run whenever any changes are pushed. maybe that's part of `push`?

Reply:
I've asked a friend who does a lot of github actions work, he's recommended changing this to `on: pull_request` entirely and setting a ~10m timeout.

the 'synchronise` type of `pull_request` would cause CI to be re-run if the branch the MR is linked to is updated.

Reply:
I have made these changes, and restricted CI to only run when a maintainer has approved it for external forks.  I think this, along with the pull_request `on:` setting and timeout gives us the best combo of security and functionality.

Reply:
Oh, for reference i was basing this on [the docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request):

```
By default, a workflow only runs when a pull_request event's activity type is opened, synchronize, or reopened.
```